ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -169,14 +169,41 @@ public boolean hasEverBeenStarted() {
     return everStarted;
   }
 
-  public void close() {
+
+  /**
+   * Stops the query without cleaning up the external resources
+   * so that it can be resumed when we call {@link #start()}.
+   *
+   * <p>NOTE: {@link TransientQueryMetadata} overrides this method
+   * since any time a transient query is stopped the external resources
+   * should be cleaned up.</p>
+   *
+   * @see #close()
+   */
+  public void stop() {
+    close(false);
+  }
```

Comment:
I'd be tempted to make this abstract and move this impl to `PersistentQuery`.

No biggie, just feels cleaner to me.

Reply:
I wanted to do that, but there's a place where `QueryMetadata` is actually being instantiated directly. Not sure if that's a bug, but I didn't want to deal with it now.

Reply:
Really? Sounds like a bug!

Reply:
Looks like this is used from the new code: for transient queries. So this should default to cleaning up internal topics.

Reply:
so we should switch this to `close(true)` as we want transient queries to delete internal state.

This will also mean we need to remove the overloaded version from `TransientQueryMetadata` and add the following overload to `PersistentQueryMetadata`

```java
@Override
public void stop() {
    close(false);
  }
```